packages/shared/src/clients/rpc/types.ts

```diff
@@ -0,0 +1,15 @@
+import { stringAsInt } from '@l2beat/shared-pure'
+import { z } from 'zod'
+
+export const RpcResponse = z.object({
+  result: z.union([z.string(), z.number(), z.record(z.string(), z.any())]),
+})
+
+export type Block = z.infer<typeof Block>
+export const Block = z.object({
+  // TODO: add support for including txs body
+  transactions: z.array(z.string()),
+  timestamp: stringAsInt(),
+  hash: z.string(),
+  number: stringAsInt(),
```

Comment:
This is not the way to do it. JSON-RPC describes a type called Quantity https://ethereum.org/en/developers/docs/apis/json-rpc/#quantities-encoding. We should be able to say `number: quantity` and also when encoding something like `quantity.encode(number)`